wagtail/tests/test_reference_index.py

```diff
@@ -500,3 +501,117 @@ def test_describe_source_field_and_on_delete_parental_key(self):
                 reference.describe_on_delete(),
                 "the advert placement will also be deleted",
             )
+
+
+# New tests for custom extract_references precedence
+
+
+class TestCustomExtractReferences(TestCase):
```

Comment:
```suggestion
class TestCustomExtractReferences(TestCase):
    """
    ... some description of the unit test class, what we are doing and why
    """
```

Might be nice though to add a simple description here.